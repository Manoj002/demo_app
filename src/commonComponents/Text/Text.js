import React from 'react';
import PropTypes from 'prop-types';
import * as COLOR from '../../constants';
import {
  DEFAULT_FONT_SIZES,
  TEXT_ALIGN,
  LINE_HEIGHT,
} from '../../constants/TextConstants';
import {TextWrapper} from './Text.styles';

const Text = ({
  color,
  fontSize,
  children,
  align,
  lineHeight,
  bold,
  marginLeft,
  marginBottom,
  marginTop,
  ...restProps
}) => (
  <TextWrapper
    color={color}
    fontSize={fontSize}
    align={align}
    lineHeight={lineHeight}
    bold={bold}
    marginLeft={marginLeft}
    marginBottom={marginBottom}
    marginTop={marginTop}
    {...restProps}>
    {children}
  </TextWrapper>
);

Text.propTypes = {
  children: PropTypes.node,
  color: PropTypes.oneOf(Object.keys(COLOR.Colors)),
  fontSize: PropTypes.oneOf(Object.values(DEFAULT_FONT_SIZES)),
  align: PropTypes.oneOf(Object.values(TEXT_ALIGN)),
  lineHeight: PropTypes.oneOf(Object.values(LINE_HEIGHT)),
  marginLeft: PropTypes.number,
  marginBottom: PropTypes.number,
  marginTop: PropTypes.number,
  bold: PropTypes.bool,
};

Text.defaultProps = {
  children: null,
  color: 'lightGray',
  fontSize: DEFAULT_FONT_SIZES.default,
  align: TEXT_ALIGN.LEFT,
  lineHeight: LINE_HEIGHT.default,
  marginLeft: 0,
  marginBottom: 0,
  marginTop: 0,
  bold: false,
};

export default Text;
