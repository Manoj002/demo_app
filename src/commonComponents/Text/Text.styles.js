import styled from 'styled-components/native';
import * as COLOR from '../../constants';
import {DEFAULT_FONT_FAMILY} from '../../constants/TextConstants';

export const TextWrapper = styled.Text`
  color: ${({color}) => COLOR.Colors[color] || COLOR.Colors.lightGray};
  font-size: ${({fontSize}) => `${fontSize}px`};
  font-family: ${DEFAULT_FONT_FAMILY};
  ${({bold}) => bold && 'font-weight: bold'};
  ${({align}) => align && `text-align: ${align}`};
  ${({lineHeight}) => lineHeight && `line-height:${lineHeight}px`};
  ${({marginLeft}) => marginLeft && `margin-left:${marginLeft}px`};
  ${({marginBottom}) => marginBottom && `margin-bottom:${marginBottom}px`};
  ${({marginTop}) => marginTop && `margin-top:${marginTop}px`};
`;

export default TextWrapper;
