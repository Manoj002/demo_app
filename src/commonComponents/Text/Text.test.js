import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Text from './Text';

describe('text Component', () => {
  it('matches snapshot', () => {
    expect(
      render(
        <Text
          bold
          fontSize={16}
          lineHeight={18}
          color="blue"
          align="center"
          marginTop={20}
          marginBottom={20}
          marginLeft={20}>
          Text Component
        </Text>,
      ),
    ).toMatchSnapshot();
  });

  it('should call onPress', () => {
    const handleTextPress = jest.fn();
    const {getByTestId} = render(
      <Text
        as="h1"
        align="right"
        testID="Simple-text"
        onPress={handleTextPress}>
        Simple text here
      </Text>,
    );
    const SimpleText = getByTestId('Simple-text');
    fireEvent.press(SimpleText);
    expect(handleTextPress).toBeCalledTimes(1);
  });
});
