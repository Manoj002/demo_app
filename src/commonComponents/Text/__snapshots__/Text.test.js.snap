// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text Component matches snapshot 1`] = `
<Text
  align="center"
  bold={true}
  color="blue"
  fontSize={16}
  lineHeight={18}
  marginBottom={20}
  marginLeft={20}
  marginTop={20}
  style={
    Array [
      Object {
        "color": "#007AFF",
        "fontFamily": "Roboto",
        "fontSize": 16,
        "fontWeight": "bold",
        "lineHeight": 18,
        "marginBottom": 20,
        "marginLeft": 20,
        "marginTop": 20,
        "textAlign": "center",
      },
    ]
  }
>
  Text Component
</Text>
`;
